File: 프로그래머스/3/150365. 미로 탈출 명령어/미로 탈출 명령어.py
def solution(n, m, x, y, r, c, k):
    down = 0
    left = 0
    right = 0
    up = 0
    
    if r - x > 0: down = r - x
    else: up = x - r
    
    if c - y > 0: right = c - y
    else: left = y - c
    
    k -= down + left + right + up
    if k < 0 or (k & 1) == 1: return "impossible"
    
    down_over = min(n - max(r, x), k // 2)
    k -= down_over * 2
    left_over = min(min(c, y) - 1, k // 2)
    k -= left_over * 2
    
    return "d"*(down + down_over) + "l"*(left + left_over) + "rl"*(k // 2) + "r"*(right + left_over) + "u"*(up + down_over)

File: 프로그래머스/3/150365. 미로 탈출 명령어/미로 탈출 명령어.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n, int m, int x, int y, int r, int c, int k) {
    int down = 0;
    int left = 0;
    int right = 0;
    int up = 0;
    
    if(r > x) down = r - x;
    else up = x - r;
    
    if(c > y) right = c - y;
    else left = y - c;
    
    k -= down + left + right + up;
    if(k < 0 || (k & 1) == 1) return "impossible";
    
    int down_over = min(n - max(r, x), k / 2);
    k -= down_over * 2;
    int left_over = min(min(c, y) - 1, k / 2);
    k -= left_over * 2;
    string rl = "";
    for(int i = 0; i < k / 2; i++) rl += "rl";
    
    return string(down + down_over, 'd') + string(left + left_over, 'l') + rl + string(right + left_over, 'r') + string(up + down_over, 'u');
}